File: new/fabric/src/main/java/net/shiroha233/roadweaver/RoadWeaverDataGenerator.java
package net.shiroha233.roadweaver;

import net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint;
import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
import net.shiroha233.datagen.ModWorldGenerator;

/**
 * Fabric 数据生成器入口
 */
public class RoadWeaverDataGenerator implements DataGeneratorEntrypoint {
    @Override
    public void onInitializeDataGenerator(FabricDataGenerator fabricDataGenerator) {
        FabricDataGenerator.Pack pack = fabricDataGenerator.createPack();
        pack.addProvider(ModWorldGenerator::new);
    }
}

File: fabric/src/main/kotlin/net/shiroha233/roadweaver/RoadWeaverDataGenerator.kt
package net.shiroha233.roadweaver

import net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint
import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator
import net.shiroha233.datagen.ModWorldGenerator

class RoadWeaverDataGenerator : DataGeneratorEntrypoint {
    override fun onInitializeDataGenerator(fabricDataGenerator: FabricDataGenerator) {
        val pack: FabricDataGenerator.Pack = fabricDataGenerator.createPack()
        pack.addProvider(::ModWorldGenerator)
    }
}
